### PR TITLE
Feature search input

### PR DESCRIPTION
**Pull request by KhalidStonehouse2001**
# Create a search input

**Ticket:** [Create a search input](https://trello.com/c/VkXRJ14B/69-create-search-input)

## Description

I have added some functions and api calls so that when something is searched in the input bar and the button has been clicked it sends an API call and responds with the following products that match that search term.

### Screenshots
Screenshot 2022-05-05 at 15 38 35

## Effected Areas

Search input, home page and input component

## Developer Checklist

I have

- [X] Made the minimum amount of change required to achieve my goal to a high standard
- [ ] Added tests that describe the change
- [ ] Created tickets for any tech debt incurred

### CODE REVIEW

File: src/components/search/index.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import { Input } from '../general/input';
+
+const StyledButton = styled.div`
+  width: 20%;
+`;
+
+export const SearchInput: React.FC = () => {
+  const [searchTerm, setSearchTerm] = useState('');
+  const handleClick = () => {
+    getSearchResults(searchTerm).then((data) => console.log(data));
```
Comment (ericamcarvalho) [inline src/components/search/index.tsx:14]: Have a look at how to use async/await syntax. It's a bit cleaner. Don't forget to handle potential errors :)

File: src/constants/api.ts [hunk 2]
```diff
@@ -3,3 +3,5 @@ export const API = `${HOST}/api/v1`;
 export const API_GET_PRODUCTS = `${API}/products`;
 export const API_GET_PRODUCT = (id: string): string =>
   `${API_GET_PRODUCTS}/${id}`;
+export const API_SEARCH_PRODUCT = (searchTerm: string) =>
+  `${API_GET_PRODUCTS}?search=${searchTerm}`;
```
Comment (ericamcarvalho) [inline src/constants/api.ts:7]: maybe one thing we need to think about - and it's something that could be asked to the rest of the team is if we want to have a limit (like limiting results only to a specific number, like 5 or 10)

Comment (ericamcarvalho) [reply src/constants/api.ts:7]: you could ask the rest of the group what they think so it can be decided as a team

**Comment by ericamcarvalho** [comment]
Just a few comments from me

**Review by isapind — COMMENTED** [review]
Looks good to me! (nothing to add to Erica's comments)

File: src/components/search/index.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+`;
+
+export const SearchInput: React.FC = () => {
+  const [searchTerm, setSearchTerm] = useState('');
+  const handleClick = async () => {
+    try {
+      const searchData = await getSearchResults(searchTerm);
+      console.log(searchData);
+    } catch (err: unknown) {
+      if (err instanceof Error) {
```
Comment (ericamcarvalho) [inline src/components/search/index.tsx:19]: Nice! Someone paid attention to the presentation :) Very well done

**Review by ericamcarvalho — APPROVED** [review]

File: src/components/search/index.tsx (hunk 3 above)
Comment (ericamcarvalho) [reply src/components/search/index.tsx:19]: Every time you do something that has a chance of going wrong, remember to handle the errors so nothing crashes

Comment (KhalidStonehouse2001) [reply src/components/search/index.tsx:19]: Thanks!!

**Pull request merged**